gtk/gtkwindow.c

```diff
@@ -673,6 +673,45 @@ startup_id_is_fake (const gchar* startup_id)
   return strncmp (startup_id, "_TIME", 5) == 0;
 }
 
+#ifdef GDK_WINDOWING_X11
+gboolean
+gtk_window_property_notify (GtkWidget *widget, GdkEventProperty *event)
+{
+  static GdkAtom unredirected;
+
+  g_return_val_if_fail (widget != NULL, FALSE);
+  g_return_val_if_fail (event != NULL, FALSE);
+
+  if (g_once_init_enter (&unredirected))
+  {
+    GdkAtom atom = gdk_atom_intern_static_string ("_GTK_WINDOW_UNREDIRECTED");
+    g_once_init_leave (&unredirected, atom);
+  }
+
+  if (event->atom == unredirected)
+    {
+      GdkToplevelX11 *toplevel = _gdk_x11_window_get_toplevel (event->window);
+      gint format, lenght;
+      GdkAtom type;
+      guchar *data;
+
+      if (gdk_property_get (event->window, event->atom, XA_CARDINAL, 0, 32,
```

Comment:
You're using `gdk_property_get()` with X11 data types. You could simply use `XGetWindowProperty()` instead. You'll also need to push and pop the error trap.